rpath hack for installed binaries.
Do not upstream.

nobtcfi on x86_64

Index: swift/cmake/modules/AddSwift.cmake
--- swift/cmake/modules/AddSwift.cmake.orig
+++ swift/cmake/modules/AddSwift.cmake
@@ -630,6 +630,7 @@ function(_add_swift_runtime_link_flags target relpath_
 
   set_property(TARGET ${target} PROPERTY BUILD_WITH_INSTALL_RPATH YES)
   set_property(TARGET ${target} APPEND PROPERTY INSTALL_RPATH "${swift_runtime_rpath}")
+  set_property(TARGET ${target} APPEND PROPERTY INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
 endfunction()
 
 # Add a new Swift host library.
@@ -969,7 +970,7 @@ function(add_swift_host_tool executable)
   endif()
 
   # Opt-out of OpenBSD BTCFI if instructed where it is enforced by default.
-  if(SWIFT_HOST_VARIANT_SDK STREQUAL "OPENBSD" AND SWIFT_HOST_VARIANT_ARCH STREQUAL "aarch64" AND NOT SWIFT_OPENBSD_BTCFI)
+  if(SWIFT_HOST_VARIANT_SDK STREQUAL "OPENBSD" AND SWIFT_HOST_VARIANT_ARCH MATCHES "aarch64|x86_64" AND NOT SWIFT_OPENBSD_BTCFI)
     target_link_options(${executable} PRIVATE "LINKER:-z,nobtcfi")
   endif()
 
@@ -991,6 +992,10 @@ function(add_swift_host_tool executable)
         TARGET ${executable}
         APPEND PROPERTY INSTALL_RPATH
           "$ORIGIN/../${extra_relative_rpath}lib/swift/host/compiler")
+      set_property(
+        TARGET ${executable}
+        APPEND PROPERTY INSTALL_RPATH
+          "${CMAKE_INSTALL_PREFIX}/lib/swift/host/compiler")
     endif()
   endif()
 
